prov/sockets/src/sock_ep_msg.c

```diff
@@ -183,36 +183,37 @@ static int sock_ep_cm_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen)
 
 static int sock_cm_send(int fd, const void *buf, size_t len)
 {
-	ssize_t ret, done = 0;
+	ssize_t ret;
+	size_t done = 0;
 
-	while ((size_t) done != len) {
+	while (done < len) {
 		ret = ofi_send_socket(fd, (const char*) buf + done,
 				      len - done, MSG_NOSIGNAL);
-		if (ret < 0) {
-			if (OFI_SOCK_TRY_SND_RCV_AGAIN(ofi_sockerr()))
-				continue;
+		if (ret >= 0) {
+			done += ret;
+		} else if (!OFI_SOCK_TRY_SND_RCV_AGAIN(ofi_sockerr())) {
 			SOCK_LOG_ERROR("failed to write to fd: %s\n",
 				       strerror(ofi_sockerr()));
 			return -FI_EIO;
 		}
```

Comment:
The convention here is to handle error in branches and let the normal path fall through. Can we keep the original flow?

Reply:
Ok, will revert the changes.